src/components/aws/services/awsSrv.js

```diff
@@ -0,0 +1,130 @@
+                   }),
+               };
+            } 
+
+            $window.AWS.config.update(config);
+         }  
+
+         function _init() {
+              _updateAwsConfig();
+         }   
+           
+         _init();
+
+          function _generateFileName() {
+             return UtilitySrv.general.createGuid();  
+          } 
+
+          function _getFile(blob, fileName) {
+              return new $window.File([blob], fileName);
+          }
+
+          function _getFilePath(prefixPath, file) {
+             return prefixPath + '/' + file.name;
+          }
+
+          function updateConfig(options, config) {
+              _updateAwsConfig(options, config);
+          }
+
+          function AwsS3(options) {
+              options = options || {};
+
+              if (!angular.isObject(options) || angular.isArray(options)) {
+                  $log.error('AwsSrv Bucket newS3: options must be an object!!');
+                  return;
+              }
+
+              this.bucketName = options.bucketName || ENV.s3InMedieBucketName || 'inmedia.zinkerz.com';
+
+              this.filesNames = [];
+
+              this.bucketInstance = new $window.AWS.S3({ 
+                   params: {
+                      Bucket: this.bucketName
+                   }
+              });
+          }
+         
+        AwsS3.prototype.upload = function(options) {
+            var deferred = $q.defer();
+            var errMsg;
+            var self = this;
+
+            if (!angular.isObject(options) || angular.isArray(options)) {
+                  errMsg = 'AwsSrv AwsS3 upload: options must be an object!! ie: { blob: blob}';
+                  $log.error(errMsg);
+                  deferred.reject(errMsg);
+                  return;
+            }
+
+            var blobOption = options.blob;
+            var fileOption = options.file;
+
+            if (!blobOption && !fileOption) {
+                  errMsg = 'AwsSrv AwsS3 upload: options must contian blob or file option!';
+                  $log.error(errMsg);
+                  deferred.reject(errMsg);
+                  return;
+            }
+            
+            var fileName = blobOption ? _generateFileName() + '.' + (options.ext || 'mp3')  : fileOption.name;
+            var file = blobOption ? _getFile(blobOption, fileName) : fileOption;
+            var filePath = _getFilePath(options.prefixPath, file);
+
+            var params = {
+                Key: filePath,
+                ContentType: file.type,
+                Body: file,
+                ACL: options.ACL || 'public-read'
+            };
+
+            this.bucketInstance.putObject(params, function(err, data) {
+                if (err) {
+                     deferred.reject(err);
+                } else {
+                     self.filesNames.push(fileName);
+
+                     deferred.resolve(data);
```

Comment:
i don't use it, it just an empty object with methods perhaps, using it as an indication that put succeed and log.debug in infra toefl